frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/StartHandler.java

```diff
@@ -259,6 +259,14 @@ public APIGatewayProxyResponseEvent handleRequest(
                             isBlockedForReauth,
                             isUserAuthenticatedWithValidProfile);
 
+            if (userStartInfo.isUpliftRequired()) {
+                var authSession = authSessionService.getSession(session.getSessionId());
+                authSession.ifPresent(
+                        authSessionItem ->
+                                authSessionService.updateSession(
```

Comment:
I'm aware that this updates the value in the Auth session at the point that _uplift is determined to be required_, not at the point that _uplift actually happens_. I think this is fine, but can we be certain that the two will never differ?